src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -330,6 +340,11 @@ def _monitor_workers_and_process_tasks_in_thread(
 
             self._processing_idx.pop(line_number)
 
+        # If the task queue is not empty, it indicates that the exit from the while loop was due to a batch run timeout.
+        # In this case, we should set is_timeout to True.
+        if not task_queue.empty():
```

Comment:
This logic is tricky, why in this case `is_timeout` is not set.

The logic here highly relies on the assumption of the implementation somewhere else, this is not a recommended practice.

Reply:
Updated.